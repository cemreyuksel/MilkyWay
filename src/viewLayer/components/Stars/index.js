import React from 'react';

import Star from '../Star';

import { useFetch } from '../../hooks';
import { STARS_URL, LOADING_MESSAGE } from '../../../dataLayer/env';

import './style.scss';

const Stars = () => {
  const [stars, loading] = useFetch(STARS_URL);

  const starItems = stars.map(({ name, color, universeId }, index) => (
    <Star key={index} name={name} color={color} universeId={universeId} />
  ));

  return (
    <>
      {loading ? (
        <h2>{`${LOADING_MESSAGE}`}</h2>
      ) : (
        <div className="star-items">{starItems}</div>
      )}
    </>
  );
};

export default Stars;
