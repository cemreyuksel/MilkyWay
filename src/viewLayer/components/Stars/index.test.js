import React from 'react';
import renderer from 'react-test-renderer';

import Stars from './index';

jest.mock('../Star', () => 'StarComponent');

jest.mock('../../hooks', () => ({
  useFetch: () => [
    [
      {
        color: 'RED',
        id: 123,
        name: 'My happy star 123',
        universeId: 333,
      },
      {
        color: 'YELLOW',
        id: 321,
        name: '321 is myy happy star',
        universeId: 33113,
      },
    ],
    false,
  ],
}));

describe('Stars component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders', () => {
    const tree = renderer.create(<Stars />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
