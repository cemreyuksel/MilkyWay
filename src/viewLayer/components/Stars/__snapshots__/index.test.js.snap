// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars component renders 1`] = `
<div
  className="star-items"
>
  <StarComponent
    color="RED"
    name="My happy star 123"
    universeId={333}
  />
  <StarComponent
    color="YELLOW"
    name="321 is myy happy star"
    universeId={33113}
  />
</div>
`;
